### DISCUSSION TITLE
Possibly adding a way in Xlights to temporarily disable Channel Updating in Layout Tab

### DISCUSSION THREAD

**Discussion by blun23** [xLightsSequencer/xLights#4385] [Ideas]
So I kinda started talking about this with a developer on Thursday night.  Now I in no way have a small show (over 1.2 million channels). Whenever I change the name of a group or model, update number of strings, update string start channel, etc., it takes a couple of minutes for xlights to make the change in my main layout.  Now if I'm on a blank layout only using a couple of props, it's instantaneous.  My best guess is that not only is it updating a start channel for the groups, models and strings, but also for every group, model and submodel that is below It in order of channel #.  No matter what, it's doing a lot in the background.  It is basically the same no matter what PC or Mac I'm using and I have some fairly beefed up computers.  

I'm curious if we could add an option to disable that until all changes are made similar to the disable rendering box in the sequence tab.  Once all changes are made, there would be a button to click to update all Group, Model and Submodel start channels.

**Comment by AzGilrock** [comment]
I didn't know if this was possible or whether it would cause issues.  A lot of things depend on knowing the mapped channels.

**Comment by cybercop23** [comment]
Are you assigning channels manually or using the visualizer? What controllers do you have? Channels are only assigned at the model level, not groups or submodels. 
Have you tried a fresh show folder and importing all of the models from the original one and see what happens? We've seen issues with some duplicates in RGBeffects file that may cause this issue. 
Maybe upload your RGBeffects and Networks files so someone can try too on your layout.

**Reply by blun23** [reply]
I'm gonna try the new layout to see if that improves.  But Submodels do have defined channel start and end channels. I've created thousands upon thousands of em on my props, so that is probably why my show lags even on stronger pc's or Macs.

![Screenshot 2024-04-05 at 9 18 18 PM](https://github.com/xLightsSequencer/xLights/assets/139409588/b84461c9-6362-4a8b-8ec6-68ef99a0e115)

**Reply by blun23** [reply]
Actually it would be pretty awesome to have the submodels export in the export models xls.  I'd love to see how many I've made.

**Comment by cybercop23** [comment]
They are just a subset of the model, but doesn't really matter since a submodel can have (and does) gaps in nodes, thus those channels are irrelevant.

**Comment by blun23** [comment]
Okay so I had this last year.  Now I've updated my Mac as many of you know to an M4 24gb but still had quite a lag doing stuff in my layout.  Still took forever to import a prop.  So I stuck with making all changes in a test folder and then importing into main layout.  So I don't know if it's attached to the recent update to detach layout from model view, but I can now do stuff in my main layout with very little delay.  It was taking almost 20 minutes to import a prop at this time last year, and it's like 10-20 seconds.  So not sure what change or changes made this happen but thank you to everyone.

**Comment by cybercop23** [comment]
@blun23 Brad, try the latest TestFlight on your orioginal show...

**Reply by blun23** [reply]
> @blun23 Brad, try the latest TestFlight on your orioginal show...

I'm on the absolute latest on the Mac and everything is kosher.  Another thing is visualizer is also super fast compared to before.  Open sequences, opening xlights (where I have save on open, but not recurse subfolders), layout, import, etc.  So far, everything that bogged down my medium ish sized show is not giving me the same issues as prior.  Whatever actually made it go faster is a mega time saver for me and anyone with larger shows.
